### DISCUSSION TITLE
Webhook event when Agent is created

### DISCUSSION THREAD

**Discussion by guiqis** [chatwoot/chatwoot#9021] [Chatwoot APIs] [answered]
Is there a Webhook endpoint that runs when an agent is added? I need to know every time an account adds an agent. How can I do it? I tested all webhooks options and nothing was triggered.

**Comment by sojan-official** [comment, marked as answer]
Hey, Chatwoot does not currently emit an account webhook event when an agent/user is created.

The webhook events available today are focused on conversations, messages, contacts, inboxes, widget events, and typing indicators. So an `agent_created` webhook is not something that can be configured from the current webhook settings.

If your goal is audit logging, the audit logs are the better place to check for this kind of admin/user-management activity.

If you need this for an external integration, provisioning workflow, IAM sync, reporting pipeline, or another real-time automation use case, this would be a good feature request. It would help to include:
- what system you want to sync agent creation into
- whether you also need `agent_updated` / `agent_deleted`
- what payload fields you need, such as agent ID, name, email, role, account ID, and timestamp
- whether the webhook should fire for invited users, confirmed users, or both

That extra context will make it easier to evaluate and design the webhook event properly.
